khmer/utils.py

```diff
@@ -18,9 +18,63 @@ def print_error(msg):
     print >>sys.stderr, msg
 
 
+def check_is_pair(record1, record2):
+    is_fastq = False
+    if hasattr(record1, 'accuracy'):
+        is_fastq = True
+        assert hasattr(record2, 'accuracy')
+
+    name1 = record1.name
+    name2 = record2.name
+
+    if is_fastq and ' ' in name1:                        # handle '@name 1:rst'
+        name1, rest1 = record1.name.split(' ', 1)
+        name2, rest2 = record2.name.split(' ', 1)
+
+        if name1 == name2 and \
+           rest1.startswith('1:') and rest2.startswith('2:'):
+            return True
+
+    elif name1.endswith('/1') and name2.endswith('/2'):  # handle name/1
+        subpart1 = name1.split('/', 1)[0]
+        subpart2 = name2.split('/', 1)[0]
+
+        assert subpart1
+        if subpart1 == subpart2:
+            return True
+
+    return False
+
+
+def broken_paired_reader(screed_iter):
```

Comment:
I had to scroll down to the script to figure out exactly what this function returns. It would be nice to have an *in situ* description. And @ctb, if you're concerned about returning 4 values, you may consider dropping the `ispair` return value and using the `read2 == None` check instead. Not a huge improvement, but 3 < 4.